backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java

```diff
@@ -753,19 +755,21 @@ public Observation convertToObservation(ConvertToObservationProps props) {
     return observation;
   }
 
-  private Coding convertToAbnormalFlagInterpretation(String resultCode) {
+  public Coding convertToAbnormalFlagInterpretation(String resultCode) {
     Coding abnormalFlag = new Coding();
 
     abnormalFlag.setSystem(ABNORMAL_FLAGS_CODE_SYSTEM);
 
-    if (resultCode.equals(DETECTED_SNOMED_CONCEPT.code())) {
+    if (ABNORMAL_SNOMEDS.keySet().contains(resultCode)) {
       abnormalFlag.setCode(ABNORMAL_FLAG_ABNORMAL.code());
       abnormalFlag.setDisplay(ABNORMAL_FLAG_ABNORMAL.displayName());
-    } else {
+    } else if (NORMAL_SNOMEDS.keySet().contains(resultCode)) {
       abnormalFlag.setCode(ABNORMAL_FLAG_NORMAL.code());
       abnormalFlag.setDisplay(ABNORMAL_FLAG_NORMAL.displayName());
+    } else {
+      log.info("Unsupported SNOMED result code: {}", Encode.forJava(resultCode));
+      return null;
```

Comment:
I think it also makes sense to adjust [this bit](https://github.com/CDCgov/prime-simplereport/blob/268ddb50c86572fc83419243cc36a45dfd4845a1/backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java#L746-L748) to not add the interpretation that the abnormal flag coding lives inside. (otherwise it will just be an empty interpretation)

Reply:
+1 agree that makes sense to omit the interpretation if it'll be empty anyway

Reply:
OO good catch -- thank you!